docs/html/_sources/Documentation/3_DesignDecisions/5_LLVMComparison/1_llvm-modules.rst.txt

```diff
@@ -0,0 +1,71 @@
+Comparing the LLVM Classes that Describe a Program
+**************************************************
+
+LLVM provides a variety of classes that  are used to describe a program, function, or instruction. These can be used
+to compare LLVM programs. In addition to that, LLVM has its own comparison program, which we may be able to use in
+part or whole. Additionally, we may be able to use techniques from code clone detectors, such as comparing the
+abstract syntax tree of an LLVM program, like the application CloneDR does (http://www.semanticdesigns
+.com/Products/Clone/HowCloneDRWorks.html). The Clang AST (http://clang.llvm.org/docs/IntroductionToTheClangAST
+.html) provides a way for us to create the AST.
+
+Text comparison is likely not the best solution as variable names and instruction order might change, without the
+code actually being very different. For this reason, basic text comparison algorithms are not being considered as
+much as other comparison metrics.
+
+This documentation begins with a list of useful functions in the LLVM API that we might use to compare LLVM
+modules. Then, there is a discussion on how the LLVM API comparison classes work and how useful they might be.
+Lastly, there is a discussion of different techniques and tools for comparing code, and howthey might be useful.
+
+------------------------------------------------------------------------------------------------------------------------
+
```

Comment:
lots of typos in this area. Not necessarily a big deal now, just mentioning it so that in the end we clean it up to look more professional.

Reply:
Fixed all these typos